lib/puppet/provider/reboot/windows.rb

```diff
@@ -2,8 +2,8 @@
 require 'open3'
 
 Puppet::Type.type(:reboot).provide :windows do
-  confine :operatingsystem => :windows
-  defaultfor :operatingsystem => :windows
+  confine :kernel => :windows
+  defaultfor :kernel => :windows
```

Comment:
I can revert it if you think it's worth doing but it seemed to make sense to leave it in for parity with the Linux provider, plus the fact that it shouldn't cause any change in behavior.  The acceptance tests validated that it works, by my understanding.  Thoughts?